mm/2s2h/Rando/ActorBehavior/ObjGrass.cpp

```diff
@@ -0,0 +1,1063 @@
+// clang-format on
+
+float roundToWholeNumber(float value) {
+    int rounded = std::round(value);
+    float fnum = static_cast<float>(rounded);
+
+    return fnum;
+}
+
+RandoCheckId IdentifyGrass(Vec3f pos) {
+    RandoCheckId randoCheckId = RC_UNKNOWN;
+    pos.x = roundToWholeNumber(pos.x);
+    pos.z = roundToWholeNumber(pos.z);
+
+    const auto& grassMap = (gPlayState->sceneId != SCENE_KAKUSIANA) ? overworldGrassMap : grottoGrassMap;
+    uint32_t key =
+        (gPlayState->sceneId != SCENE_KAKUSIANA) ? static_cast<uint16_t>(gPlayState->sceneId)
+        : (gSaveContext.save.entrance == ENTRANCE(GROTTOS, 4) || gSaveContext.save.entrance == ENTRANCE(GROTTOS, 10))
+            ? gSaveContext.respawn[RESPAWN_MODE_UNK_3].data
+            : gSaveContext.save.entrance;
+
+    auto it = grassMap.find(key);
+    if (it == grassMap.end()) {
+        return randoCheckId;
+    }
+
+    auto innerIt = it->second.find({ pos.x, pos.z });
+    if (innerIt == it->second.end()) {
+        return randoCheckId;
+    }
+
+    return innerIt->second;
+}
+
+void SpawnGrassDrop(Vec3f pos, RandoCheckId randoCheckId) {
+    CustomItem::Spawn(
+        pos.x, pos.y, pos.z, 0, CustomItem::KILL_ON_TOUCH | CustomItem::TOSS_ON_SPAWN | CustomItem::ABLE_TO_ZORA_RANG,
+        randoCheckId,
+        [](Actor* actor, PlayState* play) {
+            RandoSaveCheck& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            randoSaveCheck.eligible = true;
+        },
+        [](Actor* actor, PlayState* play) {
+            auto& randoSaveCheck = RANDO_SAVE_CHECKS[CUSTOM_ITEM_PARAM];
+            RandoItemId randoItemId = Rando::ConvertItem(randoSaveCheck.randoItemId);
+            Matrix_Scale(30.0f, 30.0f, 30.0f, MTXMODE_APPLY);
+            Rando::DrawItem(Rando::ConvertItem(randoSaveCheck.randoItemId, (RandoCheckId)CUSTOM_ITEM_PARAM), actor);
+        });
+}
+
+void Rando::ActorBehavior::InitObjGrassBehavior() {
+    COND_VB_SHOULD(VB_GRASS_DROP_COLLECTIBLE, IS_RANDO, {
+        auto actorId = static_cast<ActorId>(va_arg(args, int32_t));
+        ObjGrassElement* grassElemActor;
+        EnKusa* kusaActor;
+        ObjGrassCarry* grassCarryActor;
+        Vec3f grassIdPos = gZeroVec3f;
+        Vec3f collectiblePos = gZeroVec3f;
+
+        if (actorId == ACTOR_OBJ_GRASS) {
+            grassElemActor = va_arg(args, ObjGrassElement*);
+            grassIdPos = grassElemActor->pos;
+            collectiblePos = grassIdPos;
+        } else if (actorId == ACTOR_EN_KUSA) {
+            kusaActor = va_arg(args, EnKusa*);
+            grassIdPos = kusaActor->actor.home.pos;
+            collectiblePos = kusaActor->actor.world.pos;
+        } else if (actorId == ACTOR_OBJ_GRASS_CARRY) {
+            grassCarryActor = va_arg(args, ObjGrassCarry*);
+            collectiblePos = grassCarryActor->actor.world.pos;
+        }
+
+        RandoCheckId randoCheckId =
+            IdentifyGrass(actorId == ACTOR_OBJ_GRASS_CARRY ? grassCarryActor->grassElem->pos : grassIdPos);
```

Comment:
This ternary can be dropped by setting the pos var above:
```suggestion
        } else if (actorId == ACTOR_OBJ_GRASS_CARRY) {
            grassCarryActor = va_arg(args, ObjGrassCarry*);
            grassIdPos = grassCarryActor->grassElem->pos;
            collectiblePos = grassCarryActor->actor.world.pos;
        }

        RandoCheckId randoCheckId = IdentifyGrass(grassIdPos);
```